libnavigation-core/src/main/java/com/mapbox/navigation/core/HistoryRecordingStateChangeObserver.kt

```diff
@@ -0,0 +1,15 @@
+package com.mapbox.navigation.core
+
+import androidx.annotation.UiThread
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.trip.session.NavigationSessionState
+
+@UiThread
+interface HistoryRecordingStateChangeObserver {
```

Comment:
This should be in an internal package or tagged as experimental.

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/6074